src/sql/workbench/services/notebook/browser/notebookServiceImpl.ts

```diff
@@ -693,6 +693,7 @@ export class NotebookService extends Disposable implements INotebookService {
 			providerDescriptor.instanceReady,
 			new Promise<ISerializationProvider | undefined>((resolve, reject) => setTimeout(() => {
 				if (!providerDescriptor.instance) {
+					this._serializationProviders.delete(providerDescriptor.providerId); // Remove waiting descriptor so we don't timeout again
```

Comment:
I added these timeout changes since the Jupyter extension doesn't seem to register its kernels up front, which was causing _every_ notebook to take 30 seconds to open. With this change only the first will take 30 seconds, and then the timed out descriptor gets removed. If the kernels ever do get registered, then the descriptor will get re-added at that time, making it available for future notebooks. I'll be making further adjustments to kernel waiting behavior in future PRs.